doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,394 @@
+>>> print(solution)
+[{x: -sqrt(y)}, {x: sqrt(y)}]
+```
+
+## Using {func}`~.solve`
+
+- produces various output formats depending on the answer
+    - unless you use `dict=True` to ensure the result will be formatted as a dictionary, 
+    which we recommend if you want to extract information from the result programmatically
+- produces results which can be substituted into an expression using {meth}`~sympy.core.basic.Basic.subs`
+
+You can solve an equation using {func}`~.solve` in several ways.
+
+### Make your equation into an expression that equals zero
+
+Use the fact that any expression not in an `Eq` (equation) is automatically assumed to equal zero (0) 
+by the solving functions. You can rearrange the equation $x^2 = y$ to $x^2 - y = 0$, and 
+{func}`~.solve` that expression. This approach is convenient if you are interactively solving an 
+expression which already equals zero, or an equation that you do not mind rearranging to $expression = 0$.
+
+```py
+>>> from sympy.abc import x, y
+>>> from sympy import solve
+>>> solution = solve(x ** 2 - y, x, dict=True)
+>>> print(solution)
+[{x: -sqrt(y)}, {x: sqrt(y)}]
+```
+    
+### Put your equation into `Eq` form
+
+Put your equation into `Eq` form, then apply {func}`~.solve` to the `Eq`. 
+This approach is convenient if you are interactively solving an equation which you already have in the form of an equation,
+or which you think of as an equality.
+    
+```py
+>>> from sympy import solve, Eq
+>>> from sympy.abc import x, y
+>>> eqn = Eq(x**2, y)
+>>> solution = solve(eqn, x, dict=True)
+>>> print(solution)
+[{x: -sqrt(y)}, {x: sqrt(y)}]
+```
+    
+### Parse a string representing the equation
+
+Parse a string representing the equation into a form that SymPy can understand (`Eq` form), 
+then apply {func}`~.solve` to the parsed expression.  
+This approach is convenient if you are programmatically reading in a string. 
+We [recommend against using parsing a string if you are creating the expression yourself](https://github.com/sympy/sympy/wiki/Idioms-and-Antipatterns#strings-as-input). 
+Parsing an equation from a string requires you to use 
+[transformations](https://docs.sympy.org/dev/modules/parsing.html?highlight=parse_expr#sympy.parsing.sympy_parser.parse_expr) 
+for SymPy to interpret equals signs and create symbols from your variables.
+
+You should always include the variable to solve for if you want to extract results programmatically, 
+to ensure that SymPy solves for the desired variable. 
+To ensure SymPy will produce results in a consistent format, use `dict=True`. 
+To extract the solutions, you can iterate through the list of dictionaries:  
+    
+```py
+>>> from sympy import parse_expr, solve
+>>> from sympy.abc import x
+>>> from sympy.parsing.sympy_parser import convert_equals_signs, standard_transformations
+>>> expr = "x ** 2 = y"
+>>> parsed = parse_expr(expr, transformations=standard_transformations + (convert_equals_signs,))
```

Comment:
Probably simpler to use `transformations="all"`:
```python
In [2]: parse_expr("x**2 = y", transformations="all")
Out[2]: 
 2    
x  = y
```

Reply:
Good tip. Thanks!